src/Interpreters/executeDDLQueryOnCluster.cpp

```diff
@@ -235,18 +236,20 @@ bool maybeRemoveOnCluster(const ASTPtr & query_ptr, ContextPtr context)
         database_name = context->getCurrentDatabase();
 
     auto * query_on_cluster = dynamic_cast<ASTQueryWithOnCluster *>(query_ptr.get());
-    if (database_name != query_on_cluster->cluster)
-        return false;
-
     auto database = DatabaseCatalog::instance().tryGetDatabase(database_name);
     if (database && database->shouldReplicateQuery(context, query_ptr))
     {
+        if (!context->getSettingsRef()[Setting::ignore_on_cluster_for_replicated_database] && database_name != query_on_cluster->cluster)
+            return false;
         /// It's Replicated database and query is replicated on database level,
         /// so ON CLUSTER clause is redundant.
         query_on_cluster->cluster.clear();
         return true;
     }
 
+    if (database_name != query_on_cluster->cluster)
+        return false;
+
     return false;
```

Comment:
It looks like it will return false either way, so it can be shortened
```suggestion
    return false;
```